core_java/twitter/src/main/java/ca/jrvs/apps/twitter/controller/TwitterController.java

```diff
@@ -0,0 +1,129 @@
+package ca.jrvs.apps.twitter.controller;
+
+import ca.jrvs.apps.twitter.dao.Util.TweetUtil;
+import ca.jrvs.apps.twitter.dao.model.Tweet;
+import ca.jrvs.apps.twitter.service.Service;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.StringUtils;
+
+@org.springframework.stereotype.Controller
+public class TwitterController implements Controller {
+
+  //create logger
+  private final Logger logger = LoggerFactory.getLogger(TwitterController.class);
+
+  //class private fields
+  private static final String COORD_SEP = ":";
+  private static final String COMMA = ",";
+  private Service service;
+
+  @Autowired
+  public TwitterController(Service service) {
+    this.service = service;
+  }
+
+  /**
+   * This method is used to post a tweet given the tweet text and geo coordinates.
+   *
+   * @param args
+   * @return Tweet
+   */
+  @Override
+  public Tweet postTweet(String[] args) {
+
+    if (args.length != 3) {
+      logger.error("Invalid usage of the application");
+      throw new IllegalArgumentException(
+          "USAGE: TwitterCLIApp post \"tweet_text\" \"latitude:longitude\"");
+    }
+
+    String tweet_txt = args[1];
+    String coord = args[2];
+    String[] coordArray = coord.split(COORD_SEP);
+
+    if (coordArray.length != 2 || StringUtils.isEmpty(tweet_txt)) {
+      logger.error("Invalid number of parameters for the location");
+      throw new IllegalArgumentException(
+          "Invalid location format\nUSAGE: TwitterCLIApp post \"tweet_text\" \"latitude:longitude\"");
+    }
+
+    Double lat = null;
+    Double lon = null;
+
+    try {
+      lat = Double.parseDouble(coordArray[0]);
+      lon = Double.parseDouble(coordArray[1]);
+      logger.info("Location successfully set");
+    } catch (Exception e) {
+      logger.error("Invalid format used for the location");
+      throw new IllegalArgumentException(
+          "Invalid location format\nUSAGE: TwitterCLIApp post \"tweet_text\" \"latitude:longitude\"",
+          e);
```

Comment:
~~1. catch specific exception~~
2. remove logger.error to prevent double logging

Reply:
fix this for all similar occurrences